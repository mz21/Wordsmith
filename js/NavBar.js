var React = require('React');
var StyleSheet = require('StyleSheet');
var View = require('View');
var NavigationIcon = require('./NavigationIcon');

class NavBar extends React.Component {
  render() {
    let icons_map = [
      {
        image: require('../img/todo.png'),
        name: 'Todo'
      },
      {
        image: require('../img/add.png'),
        name: 'Add'
      },
      {
        image: require('../img/words.png'),
        name: 'Words'
      },
      {
        image: require('../img/calendar.png'),
        name: 'Calendar'
      },
      {
        image: require('../img/profile.png'),
        name: 'Profile'
      },
    ];
    const icons = icons_map.map((icon) =>
      <NavigationIcon key={icon.name} image={icon.image} name={icon.name} />
    )
    return (
      <View style={{height: 60, flexDirection: 'row', justifyContent: 'space-between'}}>
        {icons}
      </View>
    );
  }
}

var styles = StyleSheet.create({
  container: {
  }
});

module.exports = NavBar;
