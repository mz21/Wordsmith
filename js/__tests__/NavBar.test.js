import React from 'React';
import renderer from 'react-test-renderer';
import NavBar from '../NavBar';

describe('<NavBar />', function() {
  it('renders correctly', function() {
    var tree = renderer.create(<NavBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
