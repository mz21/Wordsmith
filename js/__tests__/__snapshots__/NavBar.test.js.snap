exports[`<NavBar /> renders correctly 1`] = `
<View>
  <View>
    <Image
      source={1}
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Todo
    </Text>
  </View>
  <View>
    <Image
      source={1}
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Add
    </Text>
  </View>
  <View>
    <Image
      source={1}
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Words
    </Text>
  </View>
  <View>
    <Image
      source={1}
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Calendar
    </Text>
  </View>
  <View>
    <Image
      source={1}
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Profile
    </Text>
  </View>
</View>
`;
